pybossa/core.py

```diff
@@ -118,6 +122,27 @@ def configure_app(app):
     app.url_map.strict_slashes = app.config.get('STRICT_SLASHES')
 
 
+def setup_request_profiler(app):
+    @app.before_request
+    def start_yappi():
+        if request.args.get('profileit') == '1':
+            yappi.set_clock_type('wall')
+            yappi.start()
+            g.yappiing = True
+
+    @app.after_request
+    def stop_yappi(res):
+        if request.args.get('profileit') == '1':
+            stats = yappi.get_func_stats()
+            yappi.clear_stats()
+            stream = StringIO()
+            stats.print_all(stream)
+            stream.seek(0)
+            return send_file(stream, as_attachment=True,\
+                attachment_filename = '{}.txt'.format(request.path))
```

Comment:
This appears to store the logs for each request in a separate file, with the filename being the request path. Does this path include slashes or just the terminating route name (i.e., `search.txt`, `about.txt`, etc.)?

Also, should we store the profiling metrics in GUTS or a similar cloud platform for aggregating the logs, rather than files?

One other thought, is if the filename should indicate profiling in some way. For example, `search-log.txt`, etc. Just brainstorming.

Reply:
Good points, let's change the naming. I was thinking of storing these in S3, but for now I prefer to download the results each time the parameter is passed. We can think how to store these profiler files if we want to enable it on each request.